lib/modules/manager/types.ts

```diff
@@ -15,21 +16,30 @@ export interface ManagerData<T> {
   managerData?: T;
 }
 
-export interface ExtractConfig {
+export interface ExtractConfig extends CustomExtractConfig {
   registryAliases?: Record<string, string>;
   npmrc?: string;
   npmrcMerge?: boolean;
   skipInstalls?: boolean;
 }
 
-export interface CustomExtractConfig
-  extends ExtractConfig,
-    RegexManagerTemplates {
+export interface CustomExtractConfig extends RegexManagerTemplates {
   autoReplaceStringTemplate?: string;
-  matchStrings: string[];
+  matchStrings?: string[];
```

Comment:
Isn't this field mandatory?

Reply:
It is for regex-manager. But when using the combined config keeping it mandatory causes type issues.

Reply:
If we make it mandatory, then we will need to update the tests for each manager except regex Or use a Partial<T> in the test-files. That's a logical thing to do. Will update soon.